### PR TITLE
GEOMESA-2295 Support for Kafka 1.1/2.0

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Compile version set to 2.0.0
* scala-logging bumped to 3.8.0, slf4j bumped to 1.7.25

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
I verified it works against kafka 0.9.0.1, 1.0.0 and 2.0.0

File: build/cqs.tsv [hunk 1]
```diff
@@ -231,8 +231,8 @@ org.scalatra:scalatra-auth_2.11	2.6.3	compile
 org.scalatra:scalatra-common_2.11	2.6.3	compile
 org.scalatra:scalatra-json_2.11	2.6.3	compile
 org.scalatra:scalatra-swagger_2.11	2.6.3	compile
-org.slf4j:slf4j-api	1.7.21	compile
-org.slf4j:slf4j-log4j12	1.7.21	compile
+org.slf4j:slf4j-api	1.7.25	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17145

File: build/cqs.tsv [hunk 2]
```diff
@@ -231,8 +231,8 @@ org.scalatra:scalatra-auth_2.11	2.6.3	compile
...
+org.slf4j:slf4j-log4j12	1.7.25	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17146

File: build/cqs.tsv [hunk 3]
```diff
@@ -249,8 +249,8 @@ org.apache.hadoop:hadoop-hdfs	2.6.4	provided
 org.apache.hadoop:hadoop-mapreduce-client-core	2.6.4	provided
 org.apache.hadoop:hadoop-yarn-api	2.6.4	provided
 org.apache.hadoop:hadoop-yarn-common	2.6.4	provided
-org.apache.kafka:kafka_2.11	1.0.1	provided
-org.apache.kafka:kafka-clients	1.0.1	provided
+org.apache.kafka:kafka_2.11	2.0.0	provided
+org.apache.kafka:kafka-clients	2.0.0	provided
```
Comment (elahrvivaz) [inline build/cqs.tsv]: kafka is covered under our existing works-with CQ for 0.8+

File: build/cqs.tsv [hunk 4]
```diff
@@ -50,7 +50,7 @@ com.sun.xml.bind:jaxb-core	2.2.11	compile
 com.sun.xml.bind:jaxb-impl	2.2.11	compile
 com.thoughtworks.paranamer:paranamer	2.6	compile
 com.typesafe:config	1.3.3	compile
-com.typesafe.scala-logging:scala-logging_2.11	3.1.0	compile
+com.typesafe.scala-logging:scala-logging_2.11	3.8.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=17147

File: docs/common.py [hunk 5]
```diff
@@ -92,12 +92,6 @@
 
 .. |release_source_tarball| replace:: %(url_github_archive)s/geomesa_2.11-%(release_last)s.tar.gz
 
-.. |release_kafka08_plugin| replace:: `geomesa-kafka-08-gs-plugin-2.11-%(release_last)s.tar.gz <%(url_locationtech_release)s/geomesa-kafka-08-gs-plugin_2.11/%(release_last)s/geomesa-kafka-08-gs-plugin_2.11-%(release)s-install.tar.gz>`__
-
-.. |release_kafka09_plugin| replace:: `geomesa-kafka-09-gs-plugin-2.11-%(release_last)s.tar.gz <%(url_locationtech_release)s/geomesa-kafka-09-gs-plugin_2.11/%(release_last)s/geomesa-kafka-09-gs-plugin_2.11-%(release)s-install.tar.gz>`__
-
-.. |release_kafka10_plugin| replace:: `geomesa-kafka-10-gs-plugin-2.11-%(release_last)s.tar.gz <%(url_locationtech_release)s/geomesa-kafka-10-gs-plugin_2.11/%(release_last)s/geomesa-kafka-10-gs-plugin_2.11-%(release)s-install.tar.gz>`__
```
Comment (jnh5y) [inline docs/common.py]: Is there a Kafka release bundle?

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerVersions.scala [hunk 6]
```diff
@@ -14,7 +14,7 @@ import org.apache.kafka.clients.consumer.{Consumer, ConsumerRebalanceListener}
 import org.apache.kafka.common.TopicPartition
 
 /**
-  * Reflection wrapper for KafkaConsumer methods between kafka versions 0.9 and 0.10
+  * Reflection wrapper for KafkaConsumer methods between kafka versions 0.9, 0.10, 1.0, 1.1, and 2.0
```
Comment (jnh5y) [inline geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerVersions.scala]: This seems to suggest that Kafka 0.9 - 2.0 are supported.

File: pom.xml [hunk 7]
```diff
@@ -96,9 +96,9 @@
         <hbase.guava.version>12.0.1</hbase.guava.version>
         <cassandra.version>3.0.0</cassandra.version>
         <cassandra.guava.version>18.0</cassandra.guava.version>
-        <kafka.version>1.0.1</kafka.version> <!-- also supports: 0.10.2.1, 0.9.0.1 -->
-        <zkclient.version>0.10</zkclient.version> <!-- corresponding versions: 0.10, 0.7 -->
-        <kafka.jopt.version>5.0.4</kafka.jopt.version> <!-- corresponding versions: 4.9, 3.2 -->
+        <kafka.version>2.0.0</kafka.version> <!-- also supports: 1.1.1, 1.0.2, 0.10.2.1, 0.9.0.1 -->
```
Comment (jnh5y) [inline pom.xml]: This suggests that versions 1.1.1 - 2.0.0 are supported.

Comment (elahrvivaz) [reply pom.xml]: is the comment unclear? it's meant to indicate that 2.0.0, 1.1.1, 1.0.2, 0.10.2.1 or 0.9.0.1 are supported

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerVersions.scala (hunk 6 above)
Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerVersions.scala]: yeah, that is correct

File: docs/common.py (hunk 5 above)
Comment (elahrvivaz) [reply docs/common.py]: there is, but it's not broken out by kafka version anymore - these were left over from 1.x, but aren't actually used anywhere.

File: pom.xml (hunk 7 above)
Comment (jnh5y) [reply pom.xml]: I can't read.  The lists are in the opposite orders:).

**Pull request merged**